pkg/apis/infrastructure/v1alpha2/awsmachine_types.go

```diff
@@ -74,26 +78,61 @@ type AWSMachineSpec struct {
 	// RootDeviceSize is the size of the root volume.
 	// +optional
 	RootDeviceSize int64 `json:"rootDeviceSize,omitempty"`
-
-	// AdditionalUserDataFiles specifies extra files to be passed to user_data upon creation.
-	// +optional
-	AdditionalUserDataFiles []userdata.Files `json:"additionalUserDataFiles,omitempty"`
 }
 
 // AWSMachineStatus defines the observed state of AWSMachine
 type AWSMachineStatus struct {
-	// InstanceID is the instance ID of the machine created in AWS
+	// Ready is true when the provider resource is ready.
+	Ready *bool `json:"ready,omitempty"`
```

Comment:
Remind me why this is optional?

Reply:
We went for `*bool` on `Status.InfrastructureReady`, didn't we want to keep all of them consistent? I honestly don't remember anymore why we decided to go that route, I'll need to dig some CAPI PR